Symfony/CS/Console/ConfigurationResolver.php

```diff
@@ -350,8 +355,8 @@ private function resolveConfig()
      */
     private function resolveConfigPath()
     {
-        if (is_file($this->path)) {
-            $this->config->finder(new \ArrayIterator(array(new \SplFileInfo($this->path))));
+        if (count($this->path) === 1 && is_file($this->path[0])) {
+            $this->config->finder(new \ArrayIterator(array(new \SplFileInfo($this->path[0]))));
         } elseif ($this->isStdIn) {
             $this->config->finder(new \ArrayIterator(array(new StdinFileInfo())));
         } elseif (null !== $this->path) {
```

Comment:
I would not touch `ConfigurationResolver` at all, made a subroutine in `FixCommand` to fix specific path, in that subroutine rerun only that part of `ConfigurationResolver` that need to be reconfigured